### PR TITLE
Speech details snackbar by room proximity

### PR DESCRIPTION
**Pull request by catt-stefano**
Proximity feature: passing near a room from any screen of the homeactivity, triggers a snackbar with info on the current speech in that room.

### CODE REVIEW

**Comment by rock3r** [comment]
❌  this is supposed to go against the feature branch, not `develop` ❌

File: app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+
+public class CurrentEventSnackbarService {
+
+    private static final String WHEN_DATE_TIME_FORMAT = "HH:mm";
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java]: Whiteline after this please

File: app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import net.squanchy.service.repository.EventRepository;
+import net.squanchy.support.system.CurrentTime;
+
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+
+public class CurrentEventSnackbarService {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java]: ❌  Can we just call it `CurrentEventService` and move out all the `Snackbar` stuff? It **must** not have any notions of snackbars. That should just be how you use its data later on, in the activity. Let's keep the UI and the business logic separate.

File: app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                .filter(event -> event.place().isPresent())
+                .filter(event -> event.place().get().id().equals(placeId))
+                .filter(event -> event.isDuring(currentTime.getLocalDateTime()))
+                .firstElement();
+    }
+
+    public Snackbar buildSnackbar(View view, Event event, View.OnClickListener listener) {
+        Snackbar snackbar = Snackbar.make(view, buildString(event), BaseTransientBottomBar.LENGTH_INDEFINITE);
+        snackbar.setAction(R.string.event_details, listener);
+        snackbar.setActionTextColor(view.getResources().getColor(R.color.text_inverse));
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/home/CurrentEventSnackbarService.java]: This should not be necessary if the context is themed correctly... We should look into this (in a separate PR)

File: app/src/main/java/net/squanchy/home/HomeActivity.java [hunk 4]
```diff
@@ -177,12 +183,31 @@ protected void onResume() {
...
 
+    private void showCurrentEvent(String placeId) {
+        currentEventSnackbarService.eventIn(placeId)
+                .map(this::toSnackbar)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(Snackbar::show);
+    }
+
+    private Snackbar toSnackbar(Event event) {
+        return currentEventSnackbarService.buildSnackbar(
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/home/HomeActivity.java]: :x: As mentioned in the service, let's do the whole snackbar creation here, not there

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.java [hunk 5]
```diff
@@ -88,6 +88,10 @@ public String speakersNames() {
         return speakersBuilder.toString();
     }
 
+    public boolean isDuring(LocalDateTime time) {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/domain/view/Event.java]: `isHappeningAt()`?

File: app/src/main/java/net/squanchy/support/system/CurrentTime.java [hunk 6]
```diff
@@ -1,8 +1,14 @@
...
 
+import org.joda.time.LocalDateTime;
+
 public class CurrentTime {
 
     public Long getCurrentTimestamp() {
         return System.currentTimeMillis();
     }
+
+    public LocalDateTime getLocalDateTime() {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/support/system/CurrentTime.java]: Can we rename this (and the other method too) to remove the `get` prefix?

File: app/src/main/java/net/squanchy/tweets/TweetsPageView.java [hunk 7]
```diff
@@ -41,28 +41,17 @@ public TweetsPageView(Context context, AttributeSet attrs) {
...
-    public TweetsPageView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
-        super(context, attrs, defStyleAttr, defStyleRes);
-
-        super.setOrientation(VERTICAL);
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
 
-        Activity activity = unwrapToActivityContext(context);
+        Activity activity = unwrapToActivityContext(getContext());
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/tweets/TweetsPageView.java]: ❌  why did you move this here? The injection should happen in the constructor!

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.java [hunk 8]
```diff
@@ -70,6 +54,10 @@ protected void onFinishInflate() {
         mapView = (ImageView) findViewById(R.id.venue_map);
 
         setupToolbar();
+
+        if (!isInEditMode()) {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.java]: Same here ❌  this should be in the constructor.

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -4,6 +4,7 @@
   <string name="activity_social_feed">@string/social_query</string>
   <string name="activity_schedule">Schedule</string>
   <string name="activity_favorites">Favorites</string>
+  <string name="event_details">details</string>
```
Comment (rock3r) [inline app/src/main/res/values/strings.xml]: Lowercase?

File: app/src/main/res/layout/view_page_venue.xml [hunk 10]
```diff
@@ -20,7 +21,8 @@
 
   <ScrollView
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    app:layout_behavior="@string/appbar_scrolling_view_behavior" >
```
Comment (rock3r) [inline app/src/main/res/layout/view_page_venue.xml]: To use this correctly you need to change `ScrollView` to `NestedScrollView`

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -100,4 +101,5 @@
   <string name="onboarding_error_permission_denied">We need the location permission to continue.</string>
   <string name="onboarding_error_bluetooth_denied">We need the Bluetooth to be active to continue.</string>
 
+
```
Comment (rock3r) [inline app/src/main/res/values/strings.xml]: Remove this please

**Review by rock3r — CHANGES_REQUESTED** [review]
Good work 👍  a couple of blockers, and a few notes, then we can ship.

File: app/src/main/java/net/squanchy/tweets/TweetsPageView.java (hunk 7 above)
Comment (catt-stefano) [reply app/src/main/java/net/squanchy/tweets/TweetsPageView.java]: While I'm at it, I will also fix the other 2 pageviews, schedule and favorites that already had the same flow

**Review by rock3r — APPROVED** [review]
Nice!
